src/main/java/Task/Task.java

```diff
@@ -0,0 +1,82 @@
+package Task;
+
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Abstract class representing a task with a completion status and formatting options.
+ * Hello!
+ */
+public abstract class Task {
+    protected static final DateTimeFormatter toSelfFormatter = DateTimeFormatter.ofPattern("MM-dd-yyyy HHmm");
+    protected static final DateTimeFormatter toUserFormatter = DateTimeFormatter.ofPattern("MMM dd yyyy HH:mm");
+
+    private final String task;
+    public String type;
+    private boolean completed;
+
+    /**
+     * Constructs a Task with the specified description.
+     *
+     * @param description The description of the task.
+     */
+    public Task(String description) {
+        this.task = description;
+        this.completed = false;
+        TaskList.mainTaskList.addTask(this);
+    }
+
+    /**
+     * Marks the task as completed.
+     */
+    protected void markAsDone() {
```

Comment:
since this is effectively a boolean method, it might be better to name this `setIsComplete` instead, based on coding convention!